comms/dht/src/crypt.rs

```diff
@@ -325,4 +379,163 @@ mod test {
+        assert_eq!(message, output_message);
+
+        // test for base message of base length
+        let message = vec![100u8; 984];
+        let pad_message = pad_message_to_base_length_multiple(message.as_slice());
+
+        let output_message = get_original_message_from_padded_text(pad_message.as_slice()).unwrap();
+        assert_eq!(message, output_message);
+
+        // test for empty message
+        let message: Vec<u8> = vec![];
+        let pad_message = pad_message_to_base_length_multiple(message.as_slice());
+
+        let output_message = get_original_message_from_padded_text(pad_message.as_slice()).unwrap();
+        assert_eq!(message, output_message);
+    }
+
+    #[test]
+    fn decryption_fails_if_pad_message_prepend_is_modified() {
+        let pk = CommsPublicKey::default();
+        let key = CipherKey(*chacha20::Key::from_slice(pk.as_bytes()));
+        // long text makes last test case deterministic
+        let message = "This is my secret message, keep it secret !".as_bytes().to_vec();
+        let mut encrypted = encrypt(&key, &message);
+
+        // failure in case message length prepending has been modified such that resulting
+        // length is too big to fit within pad message length
+        encrypted[size_of::<Nonce>() + 1] += 1;
+        assert!(decrypt(&key, &encrypted)
+            .unwrap_err()
+            .to_string()
+            .contains("Original length message is invalid"));
+
+        encrypted[size_of::<Nonce>() + 1] -= 1;
+
+        // failure in case message length fits whithin pad message length, but its original length has been modified
+        encrypted[size_of::<Nonce>()] -= 1;
+
+        // encrypted[size_of::<Nonce>()..size_of::<Nonce>() + le_bytes.len()].copy_from_slice(&le_bytes);
+        assert!(decrypt(&key, &encrypted).unwrap() != message);
+    }
+
+    #[test]
+    fn check_decryption_succeeds_if_pad_message_padding_is_modified() {
+        // this should not be problematic as any changes in the content of the encrypted padding, should not affect
+        // in any way the value of the decrypted content, by applying a cipher stream
+        let pk = CommsPublicKey::default();
+        let key = CipherKey(*chacha20::Key::from_slice(pk.as_bytes()));
+        let message = "My secret message, keep it secret !".as_bytes().to_vec();
+        let mut encrypted = encrypt(&key, &message);
+
+        let n = encrypted.len();
+        encrypted[n - 1] += 1;
+
+        assert!(decrypt(&key, &encrypted).unwrap() == message);
+    }
+
+    #[test]
+    fn decryption_fails_if_message_body_is_modified() {
+        let pk = CommsPublicKey::default();
+        let key = CipherKey(*chacha20::Key::from_slice(pk.as_bytes()));
+        let message = "My secret message, keep it secret !".as_bytes().to_vec();
+        let mut encrypted = encrypt(&key, &message);
+
+        encrypted[size_of::<Nonce>() + LITTLE_ENDIAN_U64_SIZE_REPRESENTATION + 1] += 1;
```

Comment:
Use wrapping_add so that if the value is 255 it can wrap to 0
```suggestion
        let b = &mut encrypted[size_of::<Nonce>() + LITTLE_ENDIAN_U64_SIZE_REPRESENTATION + 1];
        *b = b.wrapping_add(1);
```